src/main.rs

```diff
@@ -10,6 +10,16 @@ use std::collections::HashMap;
 use std::env;
 use std::sync::{Arc, Mutex};
 
+#[cfg(test)]
+mod tests {
+    use crate::detection::is_code_detected;
+
+    #[test]
+    fn is_code() {
+        assert!(is_code_detected("int main(){std::cout<<hello<<'\n'; return 0;}", 3));
```

Comment:
Since `std::` will be removed from patterns, this test should be changed too (lower threshold or add more keywords). And yeah - this code isn't valid C++ code :) `hello` -> `"hello"` at least